#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("queue_size", int_t, 0, "queue size", 100, 1, 1000)

gen.add("min_points_num", int_t, 0, "minimum points num of each cluster", 10, 1, 1000)

gen.add("sac_max_iterations", int_t, 0, "maximum iteration num for SAC segmentation", 500, 1, 5000)
gen.add("sac_distance_threshold", double_t, 0, "distance threshold for SAC segmentation", 0.005, 0.001, 0.1)
gen.add("sac_radius_limit_min", double_t, 0, "lower radius limit for SAC circle segmentation", 0.025, 0.001, 0.5)
gen.add("sac_radius_limit_max", double_t, 0, "upper radius limit for SAC circle segmentation", 0.13, 0.001, 1.0)

gen.add("box_threshold", double_t, 0, "threshold for classification as box", 0.70, 0.0, 1.0)
gen.add("circle_threshold", double_t, 0, "threshold for classification as circle", 0.30, 0.0, 1.0)


exit(gen.generate(PACKAGE, PACKAGE, "PrimitiveShapeClassifier"))
